File: java/ucotron-sdk/src/main/java/com/ucotron/sdk/UcotronRetriesExhaustedException.java
package com.ucotron.sdk;

/**
 * Exception thrown when all retry attempts have been exhausted.
 */
public class UcotronRetriesExhaustedException extends UcotronException {
    private final int attempts;

    public UcotronRetriesExhaustedException(int attempts, Throwable lastError) {
        super("All " + attempts + " retry attempts exhausted", lastError);
        this.attempts = attempts;
    }

    public int getAttempts() { return attempts; }
}

File: php/src/Exceptions/UcotronRetriesExhaustedException.php
<?php

declare(strict_types=1);

namespace Ucotron\Sdk\Exceptions;

class UcotronRetriesExhaustedException extends UcotronException
{
    public function __construct(
        public readonly int $attempts,
        string $message = '',
        ?\Throwable $previous = null,
    ) {
        $msg = $message ?: "All {$attempts} retry attempts exhausted";
        parent::__construct($msg, 0, $previous);
    }
}
